x/wasm/simulation/proposals.go

```diff
@@ -223,3 +277,130 @@ func SimulateClearAdminProposal(wasmKeeper WasmKeeper, contractSelector ClearAdm
 		)
 	}
 }
+
+type MigrateContractProposalContractSelector func(sdk.Context, WasmKeeper) sdk.AccAddress
+
+// Simulate migrate contract proposal
+func SimulateMigrateContractProposal(wasmKeeper WasmKeeper, contractSelector MigrateContractProposalContractSelector, codeSelector CodeIDSelector) simtypes.ContentSimulatorFn {
+	return func(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account) simtypes.Content {
+		ctAddress := contractSelector(ctx, wasmKeeper)
+		if ctAddress == nil {
+			return nil
+		}
+
+		codeID := codeSelector(ctx, wasmKeeper)
+		if codeID == 0 {
+			return nil
+		}
+
+		return types.NewMigrateContractProposal(
+			simtypes.RandStringOfLength(r, 10),
+			simtypes.RandStringOfLength(r, 10),
+			ctAddress.String(),
+			codeID,
+			[]byte(`{}`),
+		)
+	}
+}
+
+type SudoContractProposalContractSelector func(sdk.Context, WasmKeeper) sdk.AccAddress
+
+// Simulate sudo contract proposal
+func SimulateSudoContractProposal(wasmKeeper WasmKeeper, contractSelector SudoContractProposalContractSelector) simtypes.ContentSimulatorFn {
+	return func(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account) simtypes.Content {
+		ctAddress := contractSelector(ctx, wasmKeeper)
+		if ctAddress == nil {
+			return nil
+		}
+
+		return types.NewSudoContractProposal(
+			simtypes.RandStringOfLength(r, 10),
+			simtypes.RandStringOfLength(r, 10),
+			ctAddress.String(),
+			[]byte(`{}`),
+		)
+	}
+}
+
+// Simulate pin contract proposal
+func SimulatePinContractProposal(wasmKeeper WasmKeeper, codeSelector CodeIDSelector) simtypes.ContentSimulatorFn {
+	return func(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account) simtypes.Content {
+		codeID := codeSelector(ctx, wasmKeeper)
```

Comment:
👍 I checked the doc and the execution is idempotent so it does not really matter if it was pinned before